scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,415 @@
+        self.to_device(device)
+
+        global_samples = self._get_posterior_samples(args, kwargs, **sample_kwargs)
+        global_samples = {
+            k: v
+            for k, v in global_samples.items()
+            if k not in list(self.module.model.list_obs_plate_vars()["sites"].keys())
+        }
+
+        for k in global_samples.keys():
+            samples[k] = global_samples[k]
+
+        self.module.to(device)
+
+        return samples
+
+    def sample_posterior(
+        self,
+        num_samples: int = 1000,
+        return_sites: Optional[list] = None,
+        use_gpu: bool = False,
+        batch_size: int = 128,
+        sample_kwargs=None,
+        return_samples: bool = False,
+    ):
+        """
+        Generate samples from posterior distribution for each parameter
+        and compute mean, 5%/95% quantiles, standard deviation.
+
+        Parameters
+        ----------
+        num_samples
+            number of posterior samples to generate.
+        return_sites
+            get samples for pyro model variable, default is all variables, otherwise list variable names).
+        use_gpu
+            Use gpu?
+        sample_kwargs
+            dictionary with arguments to _get_posterior_samples (see below):
+            return_observed
+                return observed sites/variables?
+        return_samples
+            return samples in addition to sample mean, 5%/95% quantile and SD?
+
+        Returns
+        -------
+        Posterior distribution samples, a dictionary with elements as follows,
+         containing dictionaries of numpy arrays for each variable:
+            post_sample_means - mean of the distribution for each variable;
+            post_sample_q05 - 5% quantile;
+            post_sample_q95 - 95% quantile;
+            post_sample_sds - standard deviation
+            posterior_samples - samples for each variable as numpy arrays of shape `(n_samples, ...)` (Optional)
+
+        """
+
+        sample_kwargs = sample_kwargs if isinstance(sample_kwargs, dict) else dict()
+        sample_kwargs["num_samples"] = num_samples
+        sample_kwargs["return_sites"] = return_sites
+
+        # sample using minibatches (if full data, data is moved to GPU only once anyway)
+        samples = self._posterior_samples_minibatch(
+            use_gpu=use_gpu, batch_size=batch_size, **sample_kwargs
+        )
+
+        param_names = list(samples.keys())
+        results = dict()
+        if return_samples:
+            results["posterior_samples"] = samples
+
+        results["post_sample_means"] = {v: samples[v].mean(axis=0) for v in param_names}
+        results["post_sample_q05"] = {
+            v: np.quantile(samples[v], 0.05, axis=0) for v in param_names
+        }
+        results["post_sample_q95"] = {
+            v: np.quantile(samples[v], 0.95, axis=0) for v in param_names
+        }
+        results["post_sample_sds"] = {v: samples[v].std(axis=0) for v in param_names}
```

Comment:
it seems like user can supply a dict in the form `{"means": np.mean, "std": np.std}` so users can get different summary stats? and if none is supplied you give this?